src/main/java/DilemmaDetector/Simulator/FactoryWrapper.java

```diff
@@ -28,6 +28,56 @@ public boolean isPedestrian(Actor victimActor) {
         return victimActor.getEntityName().equals(victim.getOwlIndividual().getIRI().toString());
     }
 
+    public boolean isSurrounding(Actor victimActor) {
+        On_the_side surrounding = factory.getOn_the_side(victimActor.getEntity());
+        if (surrounding == null)
+            return false;
+        System.out.println(victimActor.getEntityName());
+        System.out.println(surrounding.getOwlIndividual().getIRI().toString());
+        System.out.println(victimActor.getEntityName().equals(surrounding.getOwlIndividual().getIRI().toString()));
+        return victimActor.getEntityName().equals(surrounding.getOwlIndividual().getIRI().toString());
+    }
+
+    public boolean isVehicle(Actor victimActor) {
+        Vehicle vehicle = factory.getVehicle(victimActor.getEntity());
+        if (vehicle == null)
+            return false;
+        return victimActor.getEntityName().equals(vehicle.getOwlIndividual().getIRI().toString());
+    }
+
+    public boolean isCollidableObstacle(String obstacleName )  {
+        Concrete_barrier concrete_barrier = factory.getConcrete_barrier(obstacleName);
+        Plastic_barrier plastic_barrier = factory.getPlastic_barrier(obstacleName);
+        Rock rock = factory.getRock(obstacleName);
+
+        Street_tidy street_tidy = factory.getStreet_tidy(obstacleName);
+        Pedestrian_crossing pedestrian_crossing = factory.getPedestrian_crossing(obstacleName);
+        Speed_bump speed_bump = factory.getSpeed_bump(obstacleName);
+
+        if (concrete_barrier != null || plastic_barrier != null || rock != null)
+            return true;
+        else if(street_tidy != null || pedestrian_crossing != null || speed_bump != null)
+            return false;
+        return false;
+    }
+
+    public boolean isObstacle(String obstacleName )  {
+        On_the_lane obstacle = factory.getOn_the_lane(obstacleName);
+        if (obstacle != null)
+            return true;
+
+        return false;
+    }
+
+    public boolean isObstacleOrSurrounding(String obstacleName)  {
```

Comment:
masz `isObstacle()`, masz `isSurrounding()` a `isObstaceleOrSurrounding()` musisz od nowa pisać? Zrób ora z 2 pozostałych metod, nie powtarzaj kodu, a właściwie to jak to ma tylko jedno wywołanie, to nie ma sensu tej metody tworzyć, tylko wołać `factoryWrapper.isObstacle() || factoryWrapper.isSurrounding()`.